run-tests.php

```diff
@@ -880,20 +875,19 @@ function write_information(): void
     // load list of enabled and loadable extensions
     save_text($info_file, <<<'PHP'
         <?php
-        echo str_replace("Zend OPcache", "opcache", implode(",", get_loaded_extensions()));
-        $ext_dir = ini_get("extension_dir");
+        $exts = get_loaded_extensions();
+        $ext_dir = ini_get('extension_dir');
         foreach (scandir($ext_dir) as $file) {
-            if (!preg_match('/^(?:php_)?([_a-zA-Z0-9]+)\.(?:so|dll)$/', $file, $matches)) {
-                continue;
-            }
-            $ext = $matches[1];
-            if (!extension_loaded($ext) && @dl($file)) {
-                echo ",", $ext;
+            if (preg_match('/^(?:php_)?([_a-zA-Z0-9]+)\.(?:so|dll)$/', $file, $matches)) {
+                if (@dl($matches[1])) {
```

Comment:
I think we should not load `dl()` if the `extension_loaded()`, i.e. keep this as it was before.

Reply:
this is impossible, as the `extension_loaded()` input must be the ext name, not the so/dll filename

as long as the `@dl` does not end with a fatal error (it should not), what is wrong witch this approach?

all non-fatal errors are suppressed as before and this load is done in a separate php process, thus there is no side effect to the main script

Reply:
This is for performance. Checking whether the extension is already loaded should be much faster than trying to load it again (especially on Windows). And even if some extension names have different filenames, most names match.

Reply:
```
    [29] => bz2
    [30] => 0.0010008811950684
    [31] => com_dotnet
    [32] => 0.00091910362243652
    [33] => curl
    [34] => 0.032949924468994
    [35] => dba
    [36] => 0.0010781288146973
    [37] => dl_test
    [38] => 0.00081610679626465
    [39] => enchant
    [40] => 0.01164698600769
    [41] => exif
    [42] => 0.00087594985961914
    [43] => ffi
    [44] => 0.00097990036010742
    [45] => fileinfo
    [46] => 0.0011739730834961
    [47] => ftp
    [48] => 0.00086712837219238
    [49] => gd
    [50] => 0.0023589134216309
    [51] => gettext
    [52] => 0.00079488754272461
    [53] => gmp
    [54] => 0.00092196464538574
    [55] => imap
    [56] => 0.0031580924987793
    [57] => intl
    [58] => 0.040650129318237
    [59] => mbstring
    [60] => 0.0014688968658447
    [61] => mysqli
    [62] => 0.0011990070343018
    [63] => odbc
    [64] => 0.0031149387359619
    [65] => pdo_firebird
    [66] => 0.0040609836578369
    [67] => pdo_mysql
    [68] => 0.0008399486541748
    [69] => pdo_odbc
    [70] => 0.0032060146331787
    [71] => pdo_pgsql
    [72] => 0.005363941192627
    [73] => pdo_sqlite
    [74] => 0.0039880275726318
    [75] => pgsql
    [76] => 0.0011382102966309
    [77] => shmop
    [78] => 0.00078010559082031
    [79] => snmp
    [80] => 0.087687969207764
    [81] => soap
    [82] => 0.0021140575408936
    [83] => sockets
    [84] => 0.0011708736419678
    [85] => sodium
    [86] => 0.011409997940063
    [87] => sqlite3
    [88] => 0.0010068416595459
    [89] => sysvshm
    [90] => 0.00088095664978027
    [91] => tidy
    [92] => 0.0016140937805176
    [93] => xsl
    [94] => 0.0010459423065186
    [95] => zend_test
    [96] => 0.00094985961914062
    [97] => zip
    [98] => 0.0012450218200684
```

performance isn't a problem, but I have found a problem in PHP, will report in a ticket